kratos/spatial_containers/specialized_spatial_search.cpp

```diff
@@ -0,0 +1,353 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |  (   | |  (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "spatial_containers/specialized_spatial_search.h"
+#include "spatial_containers/spatial_containers.h"
+
+namespace Kratos
+{
+
+template<>
+void PointObject<Node<3>>::UpdatePoint()
+{
+    noalias(this->Coordinates()) = mpObject->Coordinates();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void PointObject<Condition>::UpdatePoint()
+{
+    noalias(this->Coordinates()) = mpObject->GetGeometry().Center().Coordinates();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void PointObject<Element>::UpdatePoint()
+{
+    noalias(this->Coordinates()) = mpObject->GetGeometry().Center().Coordinates();
+}
+
+template class PointObject<Node<3>>;
+template class PointObject<Condition>;
+template class PointObject<Element>;
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<SpatialContainer TSearchBackend>
+void SpecializedSpatialSearch<TSearchBackend>::SearchElementsInRadiusExclusive(
+    const ElementsContainerType& rStructureElements,
+    const ElementsContainerType& rInputElements,
+    const RadiusArrayType& rRadius,
+    VectorResultElementsContainerType& rResults,
+    VectorDistanceType& rResultsDistance
+    )
+{
+    // Defining the point type for the search
+    using PointType = PointObject<Element>;
+    using PointTypePointer = typename PointType::Pointer;
+    using PointVector = std::vector<PointTypePointer>;
+    using PointIterator = PointVector::iterator;
+    using DistanceIterator = std::vector<double>::iterator;
+
+    // Defining the PointVector
+    PointVector points = PrepareSearch(rStructureElements, rInputElements, rResults, rResultsDistance);
+
+    // Defining the search structure
+    if constexpr (TSearchBackend == SpatialContainer::KDTree) {
```

Comment:
`KDTree` is misleading here because you only support octrees

Reply:
That's not true, I support all three used